girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -219,8 +219,9 @@ def finalizeUpload(self, upload, file):
 
         # Store the hash in the upload so that deleting a file won't delete
         # this file
-        upload['sha512'] = hash
-        self.model('upload').update({'_id': upload['_id']}, update={'$set': {'sha512': hash}})
+        if '_id' in upload:
```

Comment:
My reasoning, probably faulty, was that we don't save an upload model when we import files, so not saving them when uploading would be okay.

We save the upload multiple times (which is part of why this is a significant speed up), and in none of them do we check to see if the process is cancelled, though a handler could throw an exception.

The user quota plugin does use the initial save to check if there is enough space, but, since multiple uploads can be started concurrently, it also checks when the upload is finalized (and that event still occurs).  Perhaps wrongly, the user quota counts imported files the same as uploaded files.